python/pyspark/ml/feature.py

```diff
@@ -57,8 +58,8 @@ class Binarizer(JavaTransformer, HasInputCol, HasOutputCol):
     .. versionadded:: 1.4.0
     """
 
-    threshold = Param(Params._dummy(), "threshold",
-                      "threshold in binary classification prediction, in range [0, 1]")
+    threshold = FloatParam(Params._dummy(), "threshold",
```

Comment:
The range information was incorrect. There are Python unit tests that set this value to 2.0. Further, if you check the binarizer threshold param in Scala, there is no range validation check. I do believe there is no real range restriction on this parameter.